server/controllers/mentionController.js

```diff
@@ -2,12 +2,35 @@
 const Mention = require("../models/mentionModel");
 
 // @desc    Get Mentions
-// @route   GET /api/mentions/
+// @route   GET /api/mentions?platforms=xxx&keywords=xxx&sortbydate=xxx&sortbypop=xxx
 // @access  Private
 const getMentions = async (req, res) => {
   try {
-    const mentions = await Mention.find({});
-    res.json(mentions);
+    // const { user } = req;
+    const { platforms, keywords, sortbydate, sortbypop } = req.query;
+    // Test HTTP request URL :
+    // http://localhost:3001/api/mentions?platforms=reddit,twitter,facebook&keywords=bitcoin&sortbydate=asc&sortbypop=asc
+    const platformsArray = platforms.split(",");
+    const keywordsArray = keywords.split(",");
+
+    // Fetching Mentions pertaining to selected platforms
+    const fetchMentions = async (array) => {
+      const allMentions = [];
+      for (const platform of array) {
+        const mentions = await Mention.find({ platform });
+        allMentions.push(mentions);
+      }
+      return allMentions;
+    };
```

Comment:
You can optimize this by handling the promises in parallel. Check out the docs on [Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).

Reply:
I just changed the code to use Promise.all ... were you referring to that option ?